sql_joins_demystified/sql_joins_demystified.md

```diff
@@ -0,0 +1,589 @@
+| item_id (PK)  | item_name  |
+| :--------- | :--------- |
+| 15 | distilled water |
+| 178 | napkins |
+| 210 | orane juice |
+| 97 | peanut butter |
+| 108 | pistachios  |
+| 233 | plain bagels  |
+
+**`orders`**
+
+*Note: here we want to allow the `order_id` to repeat as many times as necessary to include all items, so it's not a primary key.* 
+
+<!-- data-type="none" class="tight-table" -->
+| order_id | item_id |
+| :--------- | :--------- |
+| 23125 | 210 |
+| 23125 | 108 |
+| 41320 | 97 |
+| 41320 | 233 |
+| 41320 | 210  |
+| 53011 | 178 |
+| 53011 | 15 |
+| 14123 | 108 |
+| 14123 | 210 |
+| 14123 | 97 |
+
+In the two table option, we only have to find "orane juice" in a single column, "item\_name", and it occurs once only (this is because this id number is a primary key, which means it can only occur once).
+
+The theory of how to separate data for top speed and efficiency is complex. However, it's enough for now to understand that data in SQL databases are fragmented in a process called normalization, to help reduce needless repetition, simplify data, and improve performance.
+
+### Quiz: Why is SQL Data In So Many Tables?
+
+Why is data normalized (carefully fragmented to reduce repetition and inefficiency) in SQL?  Select all the correct answers.
+
+[[ ]] SQL has a limit on number of columns permitted in a table, and must break up data in order to keep tables smaller
+[[X]] Because data is often one-to-many, storing data in a single table is inefficient
+[[X]] Normalizing data makes correcting or changing data simpler and less prone to error
+[[ ]] Normalization is a holdover from when most databases were business-related, but it not necessary in biomedical research
+[[?]] There are multiple correct answers!
+*********
+
+<div class = "answer">
+
+The reason for normalization isn't that SQL lacks the capacity for very wide tables with lots of columns.  Rather, the kind of rich data that is stored in SQL databases simply isn't right for storage in a single table.  There are many one-to-many relationships in data, and this means that using multiple tables, one for each kind of concept (like "address" or "order" or "encounter") makes sense.  There is also a lot of repetition in data.  For example, multiple patients might live at the same address, and multiple patients are prescribed the same medication.  Multiple orders each contain the same popular product.  This repetition means that it's much easier to correct or change data if we try to put as much data as possible into a single table. However, when a normalized model is used, and data is fragmented into concepts, things get easier.  Instead of correcting a product name thousands of times, once for each order that product appears on, we only have to correct it once, in the "product" table.
+
+</div>
+
+***********
+
+## Forming a JOIN 
+
+As we established earlier, JOINS are used to combine rows from two (or more) tables, based on some set of columns they have in common.
```

Comment:
Should it be JOINs with a lowercase s to make it plural?

Reply:
I'd also like to add a definition of JOIN here the way we (I) added a definition of Normalization to the start of the previous section.

Reply:
Yeah I know... we need a style guide for stuff like this!